odoo/fields.py

```diff
@@ -2379,8 +2379,9 @@ def write(self, records, value):
         cache_value = self.convert_to_cache(value if self.related else new_value, records)
         records.env.cache.update(records, self, [cache_value] * len(records))
 
-    def _image_process(self, value):
-        if self.readonly and not self.max_width and not self.max_height:
+    def _image_process(self, value, model):
+        if (self.readonly and not self.max_width and not self.max_height) or \
+            model.env['ir.config_parameter'].sudo().get_param('global.no_quality_optimization'):
```

Comment:
Something to investigate: create a helper method like `_is_optimization_disabled()`, call it here.
Make it check the global ICP by default.
Based on context/request in web_editor (or website, you decide, we will see from there), make it also consider the other ICP.